### ISSUE TITLE
Ordering of seaching policies by id should favour built in policies.

### ISSUE THREAD

**Issue by Nepomuceno** [hashicorp/terraform-provider-azurerm#18337]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.9

### AzureRM Provider Version

3.22.0

### Affected Resource(s)/Data Source(s)

azurerm_policy_definition, azurerm_policy_set_definition

### Terraform Configuration Files

```hcl
terraform {
}

provider "azurerm" {
  features {}
}

# NIST 5.3 policy.
data "azurerm_policy_set_definition" "policy_set" {
  name = "179d1daa-458f-4e47-8086-2a68d0d6c38f"
}

data "azurerm_policy_definition" "for_each_pol" {
    count = length(data.azurerm_policy_set_definition.policy_set.policy_definition_reference)
    name = data.azurerm_policy_set_definition.policy_set.policy_definition_reference[0].reference_id
}
```

### Debug Output/Panic Output

```shell
Takes over 2 minutes to do the data reads becasue of the throtolling limits.
```

### Expected Behaviour

The search on hte built in should be done before the search in the custome since custom policies are normally on a much smaller number them the built in one that way there is a biggest change of you find the policies on the built in them on the custom.

### Actual Behaviour

The plan takes  more than 2 minutes

### Steps to Reproduce

terraform plan

**Comment by wuxu92** [comment]
thanks for your feedback, since you have already created a PR, let's wait for hashi's response.

**Comment by tombuildsstuff** [comment]
Fixed via #18338
